curated_transformers/models/pytorch/curated_transformer.py

```diff
@@ -25,6 +25,10 @@ def __init__(self, encoder: CuratedEncoderT) -> None:
         super().__init__()
         self.curated_encoder = encoder
 
+        assert isinstance(
+            encoder, (AlbertEncoder, BertEncoder, RobertaEncoder)
+        ), f"Unexpected curated encoder type '{type(encoder)}'"
+
```

Comment:
I am not sure if we should go here? The type already states which encoders are allowed. If we really want to do this, we should get the list of classes from the typevar constraint, so that we don't have to maintain it in two places.

Reply:
Ah, I wasn't aware that we could query types from the `TypeVar` instance.

The original thought was to catch the usage of an unsupported encoder as early as possible as a pre-condition (since the initialization of `CuratedTransformer` happens before the initialization callbacks) and have a single source of truth, i.e., this module.

We can do away with this check if you don't think it's necessary. On second thought, we don't consistently perform runtime type checking in this way elsewhere after all.

Reply:
> On second thought, we don't consistently perform runtime type checking in this way elsewhere after all.

Indeed, so I think we should just remove it for now. The type system reject these things and worst case it triggers the later exception.